lib/kamal/cli/accessory.rb

```diff
@@ -226,19 +226,17 @@ def remove_service_directory(name)
 
   private
     def with_accessory(name)
-      if accessory = KAMAL.accessory(name)
-        yield accessory, accessory_hosts(accessory)
+      if KAMAL.config.accessory(name)
+        yield KAMAL.accessory(name), accessory_hosts(accessory)
       else
-        error_on_missing_accessory(name)
+        raise ArgumentError, error_on_missing_accessory(name)
```

Comment:
Why did you decide to raise ArgumentError instead of using `Thor#error` method?

Reply:
Reverted it back.